core-bundle/assets/controllers/tinymce-controller.js

```diff
@@ -2,6 +2,17 @@ import { Controller } from '@hotwired/stimulus';
 
 export default class extends Controller {
     connect() {
+        // Work around a bug in Safari, where transitioning to a new context
+        // leads to disconnect() being called before connect(). If the
+        // element id is identical - which is the case when saving a record -
+        // this messes up the initialization of the editor. In order to prevent
+        // this from happening we delay the execution until the call stack has
+        // cleared, and we are sure all microtasks (i.e. disconnect() calls)
+        // did run.
+        setTimeout(() => this._connect(), 0);
```

Comment:
```suggestion
        queueMicrotask(() => this._connect());
```
Did you try if queueing a microtask would be enough delay to fix the issue as well?

Reply:
Tested it and it seems to work as well! See 895b813